compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -452,21 +469,6 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
 
         // Try Levenshtein algorithm.
         let typo_sugg = self.lookup_typo_candidate(path, ns, is_expected, span);
-        // Try context-dependent help if relaxed lookup didn't work.
-        if let Some(res) = res {
-            if self.smart_resolve_context_dependent_help(
-                &mut err,
-                span,
-                source,
-                res,
-                &path_str,
-                &fallback_label,
-            ) {
-                // We do this to avoid losing a secondary span when we override the main error span.
-                self.r.add_typo_suggestion(&mut err, typo_sugg, ident_span);
-                return (err, candidates);
-            }
-        }
```

Comment:
This block of code should stay here, all the test changes happened because it had been changed from running unconditionally to running only for a subset of calls.